base/dsfs/dataset_test.go

```diff
@@ -859,3 +864,203 @@ func TestGetDepth(t *testing.T) {
 		}
 	}
 }
+
+func GenerateDataset(b *testing.B, sampleSize int, format string) (int, *dataset.Dataset) {
```

Comment:
Refactored this to include some of the nicer aspects of the dataset/dsio API:

```go
func GenerateDataset(b *testing.B, sampleSize int, format string) (int, *dataset.Dataset) {
	ds := &dataset.Dataset{
		Commit: &dataset.Commit{
			Timestamp: time.Date(2017, 1, 1, 1, 0, 0, 0, time.UTC),
			Title:     "initial commit",
		},
		Meta: &dataset.Meta{
			Title: "performance benchmark data",
		},
		Structure: &dataset.Structure{
			Format: format,
			FormatConfig: map[string]interface{}{
				"headerRow":  true,
				"lazyQuotes": true,
			},
			Schema: map[string]interface{}{
				"type": "array",
				"items": map[string]interface{}{
					"type": "array",
					"items": []interface{}{
						map[string]interface{}{"title": "uuid", "type": "string"},
						map[string]interface{}{"title": "ingest", "type": "string"},
						map[string]interface{}{"title": "occurred", "type": "string"},
						map[string]interface{}{"title": "raw_data", "type": "string"},
					},
				},
			},
		},
	}

	gen, err := generate.NewTabularGenerator(ds.Structure)
	if err != nil {
		b.Fatalf("creating generator: %s", err.Error())
	}
	defer gen.Close()

	bodyBuffer := &bytes.Buffer{}
	w, err := dsio.NewEntryWriter(ds.Structure, bodyBuffer)
	if err != nil {
		b.Fatalf("creating entry writer: %s", err.Error())
	}

	for i := 0; i < sampleSize; i++ {
		ent, err := gen.ReadEntry()
		if err != nil {
			b.Fatalf("reading generator entry: %s", err.Error())
		}
		w.WriteEntry(ent)
	}
	if err := w.Close(); err != nil {
		b.Fatalf("closing writer: %s", err)
	}

	fileName := fmt.Sprintf("body.%s", ds.Structure.Format)
	ds.SetBodyFile(qfs.NewMemfileReader(fileName, bodyBuffer))

	return bodyBuffer.Len(), ds
}
```

This function is a great mini-program! High level feedback:
* skip the unmarshaling step, just make go types. Verbose? yes. Easier to plug variables tho, and prevents potential marshaling bugs from creeping into the test
* dsio package tries to let you abstract away from the data format
* use `Fatalf` in setup functions. Tests should fail on the spot if setup fails
* use `defer` statements near the opening call site to ensure stuff closes in all cases
* always handle the error from `entryWriter.Close`, that's where stuff usually breaks
* a `*bytes.Buffer` is a `io.ReadWriter`, we have methods for creating files from readers

Upside: ran this before & after with the same bench results, which  proves to me this PR isolates setup from benchmarks properly!

Reply:
Oh boy,  this is great stuff! Thx!